src/FffGcodeWriter.cpp

```diff
@@ -2553,44 +2557,45 @@ void FffGcodeWriter::addPrimeTower(const SliceDataStorage& storage, LayerPlan& g
 
 void FffGcodeWriter::finalize()
 {
-    double print_time = gcode.getSumTotalPrintTimes();
+    Duration print_time = gcode.getSumTotalPrintTimes();
     std::vector<double> filament_used;
     std::vector<std::string> material_ids;
     std::vector<bool> extruder_is_used;
-    for (int extr_nr = 0; extr_nr < getSettingAsCount("machine_extruder_count"); extr_nr++)
+    for (size_t extr_nr = 0; extr_nr < Application::getInstance().current_slice->scene.extruders.size(); extr_nr++)
     {
         filament_used.emplace_back(gcode.getTotalFilamentUsed(extr_nr));
         material_ids.emplace_back(gcode.getMaterialGUID(extr_nr));
         extruder_is_used.push_back(gcode.getExtruderIsUsed(extr_nr));
     }
     std::string prefix = gcode.getFileHeader(extruder_is_used, &print_time, filament_used, material_ids);
-    if (CommandSocket::isInstantiated())
+    if (!Application::getInstance().communication->isSequential())
```

Comment:
It's always been like this because the architecture is a bit difficult if we want to move this check to the communication class. You'd store a temporary g-code prefix and override it with more details once they become available. But then you also need to make sure to send the prefix once and only once when you flush the g-code. Warrants a live discussion if you're interested.

Reply:
It can be done either way. I know that it was like this for a while, but maybe keep the information about the prefix in the Communication class makes more sense, since the prefix can be different for every kind of communication or they do different things when sending the GCode prefix.

At this moment, we can keep it and I don't think it is important to be done in the refactor and it can be addressed later on.